pkg/handler/snapshots.go

```diff
@@ -206,3 +224,211 @@ func (sh *SnapshotHandler) getRepoConfigurationFile(c echo.Context) error {
 
 	return c.String(http.StatusOK, repoConfigFile)
 }
+
+// DeleteSnapshot godoc
+// @summary 		Delete a snapshot
+// @ID				deleteSnapshot
+// @Description     This enables deleting a specific snapshot.
+// @Tags			snapshots
+// @Param  			repo_uuid path string true "Repository UUID."
+// @Param  			snapshot_uuid path string true "Snapshot UUID."
+// @Success			204 "Snapshot was successfully deleted"
+// @Failure      	400 {object} ce.ErrorResponse
+// @Failure     	401 {object} ce.ErrorResponse
+// @Failure      	404 {object} ce.ErrorResponse
+// @Failure      	500 {object} ce.ErrorResponse
+// @Router			/repositories/{repo_uuid}/snapshots/{snapshot_uuid} [delete]
+func (sh *SnapshotHandler) deleteSnapshot(c echo.Context) error {
+	_, orgID := getAccountIdOrgId(c)
+	repoUUID := c.Param("repo_uuid")
+	snapshotUUID := c.Param("snapshot_uuid")
+
+	repoSnaps, err := sh.isDeleteAllowed(c, orgID, repoUUID, snapshotUUID)
+	if err != nil {
+		var errResp ce.ErrorResponse
+		if errors.As(err, &errResp) {
+			return errResp
+		}
+		return ce.NewErrorResponse(http.StatusBadRequest, "Error deleting snapshots", err.Error())
+	}
+
+	_, err = sh.DaoRegistry.Snapshot.Fetch(c.Request().Context(), snapshotUUID)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching snapshot", err.Error())
+	}
```

Comment:
Since we're fetching all the snapshots for the repo, I'm not sure that fetching the individual snapshot is necessary. What do you think?

Reply:
you're right, if the snapshotUUID(s) are present in the fetched snapshots for that repo, we don't have to refetch them again 👍🏼